plugins/source/aws/client/testing.go

```diff
@@ -24,77 +19,39 @@ type TestOptions struct {
 	t.Helper()
 	ctrl := gomock.NewController(t)
 	l := zerolog.New(zerolog.NewTestWriter(t)).Output(
 		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.StampMicro},
 	).Level(zerolog.DebugLevel).With().Timestamp().Logger()
 
-	newTestExecutionClient := func(ctx context.Context, logger zerolog.Logger, spec specs.Source, _ source.Options) (schema.ClientMeta, error) {
-		var awsSpec Spec
-		if err := spec.UnmarshalSpec(&awsSpec); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal aws spec: %w", err)
-		}
-		awsSpec.SetDefaults()
-		awsSpec.UsePaidAPIs = true
-		awsSpec.TableOptions = &testOpts.TableOptions
-		c := NewAwsClient(l, nil, &awsSpec)
-		services := builder(t, ctrl)
-		services.Regions = []string{"us-east-1"}
-		c.ServicesManager.InitServicesForPartitionAccount("aws", "testAccount", services)
-		c.Partition = "aws"
-		return &c, nil
+	var awsSpec Spec
+	awsSpec.SetDefaults()
+	awsSpec.UsePaidAPIs = true
+	awsSpec.TableOptions = &testOpts.TableOptions
+	c := NewAwsClient(l, &awsSpec)
+	services := builder(t, ctrl)
+	services.Regions = []string{"us-east-1"}
+	c.ServicesManager.InitServicesForPartitionAccount("aws", "testAccount", services)
+	c.Partition = "aws"
+	tables := schema.Tables{table}
+
+	if err := transformers.TransformTables(tables); err != nil {
+		t.Fatal(err)
 	}
 
-	p := source.NewPlugin(
-		table.Name,
-		version,
-		[]*schema.Table{
-			table,
-		},
-		newTestExecutionClient)
-	p.SetLogger(l)
-	source.TestPluginSync(t, p, specs.Source{
-		Name:         "dev",
-		Path:         "cloudquery/dev",
-		Version:      version,
-		Tables:       []string{table.Name},
-		Destinations: []string{"mock-destination"},
-	}, source.WithTestPluginAdditionalValidators(validateTagStructure))
-}
-func extractTables(tables schema.Tables) schema.Tables {
-	result := make(schema.Tables, 0)
-	for _, table := range tables {
-		result = append(result, table)
-		result = append(result, extractTables(table.Relations)...)
+	sc := scheduler.NewScheduler(&c, scheduler.WithLogger(l))
+	messages, err := sc.SyncAll(context.Background(), tables)
+	if err != nil {
+		t.Fatal(err)
 	}
-	return result
-}
 
-func validateTagStructure(t *testing.T, plugin *source.Plugin, resources []*schema.Resource) {
-	for _, table := range extractTables(plugin.Tables()) {
-		t.Run(table.Name, func(t *testing.T) {
-			for _, column := range table.Columns {
-				if column.Name != "tags" {
-					continue
-				}
-				if !arrow.TypeEqual(column.Type, types.ExtensionTypes.JSON) {
-					t.Fatalf("tags column in %s should be of type JSON", table.Name)
-				}
-				for _, resource := range resources {
-					if resource.Table.Name != table.Name {
-						continue
-					}
-					value := resource.Get(column.Name)
-					var tags map[string]any
-					if err := json.Unmarshal(value.(*scalar.JSON).Value, &tags); err != nil {
-						t.Fatalf("failed to unmarshal tags column %s: %v", value.(*scalar.JSON).Value, err)
-					}
-				}
-			}
-		})
+	inserts := messages.InsertMessage()
```

Comment:
Shouldn't we also be checking that the table migration messages are sent before the insert messages? Or is that being tested somewhere else? (Feels like we probably don't have to test that in every plugin, but we don't have to test it somewhere :) )